src/solvers/dgsem_tree/dg_2d_subcell_limiters.jl

```diff
@@ -101,34 +123,44 @@ end
 @inline function subcell_limiting_kernel!(du, u,
                                           element,
                                           mesh::Union{TreeMesh{2}, StructuredMesh{2}},
-                                          nonconservative_terms::False, equations,
+                                          nonconservative_terms, equations,
                                           volume_integral, limiter::SubcellLimiterIDP,
                                           dg::DGSEM, cache)
     @unpack inverse_weights = dg.basis
     @unpack volume_flux_dg, volume_flux_fv = volume_integral
 
     # high-order DG fluxes
-    @unpack fhat1_threaded, fhat2_threaded = cache
-
-    fhat1 = fhat1_threaded[Threads.threadid()]
-    fhat2 = fhat2_threaded[Threads.threadid()]
-    calcflux_fhat!(fhat1, fhat2, u, mesh,
-                   nonconservative_terms, equations, volume_flux_dg, dg, element, cache)
-
+    @unpack fhat1_L_threaded, fhat1_R_threaded, fhat2_L_threaded, fhat2_R_threaded = cache
+
+    fhat1_L = fhat1_L_threaded[Threads.threadid()]
+    fhat1_R = fhat1_R_threaded[Threads.threadid()]
+    fhat2_L = fhat2_L_threaded[Threads.threadid()]
+    fhat2_R = fhat2_R_threaded[Threads.threadid()]
+    @trixi_timeit timer() "calcflux_fhat!" begin
+        calcflux_fhat!(fhat1_L, fhat1_R, fhat2_L, fhat2_R, u, mesh,
+                       nonconservative_terms, equations, volume_flux_dg, dg, element,
+                       cache)
+    end
```

Comment:
```suggestion
calcflux_fhat!(fhat1_L, fhat1_R, fhat2_L, fhat2_R, u, mesh,
               nonconservative_terms, equations, volume_flux_dg, dg, element,
               cache)
```
In the past, I had some issues using a timer here when using more threads, since this function is called in parallel per element. Did you try this and got no problems?

Also for the other timers.

Reply:
True.. This was just for debugging in the other PR. Thanks!